contracts/ROSCA.sol

```diff
@@ -130,10 +150,47 @@ contract ROSCA {
 
       currentRound++;
       LogStartOfRound(currentRound);
+    } else if (currentRound == membersAddresses.length) {
+        endOfROSCA = true;
     }
   }
 
   /**
+   * try to send the fund to the opt_destination from source account
+   */
+  function sendFund(address source,address opt_destination) internal returns(bool success){
+    uint amountToWithdraw = members[source].pendingWithdrawal  ;  // use a temporary variable to avoid the receiver calling the withdraw function again before
+    if(this.balance < amountToWithdraw) amountToWithdraw = this.balance;
+    members[source].pendingWithdrawal = members[source].pendingWithdrawal - amountToWithdraw;                      // send() complete its task
+    if (!opt_destination.send(amountToWithdraw)) {   // if the send() fails, put the allowance back to its original place
+      // No need to call throw here, just reset the amount owing
+      members[source].pendingWithdrawal = amountToWithdraw;
+      return false;
+    } else {
+      LogFundsWithdrawal(source, amountToWithdraw, opt_destination);
+      return true;
+    }
+  }
+
+  /**
+   * cleanUp is to make sure that fees are send to where they are owed and
+   * make sure contract doesn't hold any ether after it ends.
+   *
+   */
+  function cleanUp() {
```

Comment:
hmmm what is the incentive for anyone to call this? I'm afraid no one will want to waste gas on that.
Moreover, since participants' addresses may be contracts themselves, this call could take a LOT of gas.
I suggest removing this function, letting participants draw their own money, and letting WeTrust draw its money (we'll have the frontend do that). This way no one needs to worry about gas of other people's contracts.